tests/units/txBatch/bulker.optimisation.test.ts

```diff
@@ -0,0 +1,303 @@
+        ? BASE_AMOUNT
+        : BASE_AMOUNT.mul(2);
+
+      if (reverse && relation === OperationRelation.bigger) {
+        mock.userMarketsData[Underlying.dai].walletBalance = DELTA;
+      }
+      break;
+    }
+    case TransactionType.borrow: {
+      mock.userMarketsData[Underlying.dai].scaledCollateral =
+        parseUnits("1000"); //Borrow capacity shouldn't be a problem
+      mock.marketsData[Underlying.weth].poolLiquidity = reverse
+        ? BASE_AMOUNT
+        : BASE_AMOUNT.mul(2);
+
+      if (reverse && relation === OperationRelation.bigger) {
+        mock.userMarketsData[Underlying.weth].walletBalance = DELTA;
+        mock.userMarketsData[Underlying.weth].scaledBorrowOnPool = DELTA;
+      }
+      break;
+    }
+    case TransactionType.withdraw: {
+      mock.userMarketsData[Underlying.weth].scaledSupplyOnPool = reverse
+        ? BASE_AMOUNT
+        : BASE_AMOUNT.mul(2);
+
+      if (reverse && relation === OperationRelation.bigger) {
+        mock.userMarketsData[Underlying.weth].walletBalance = DELTA;
+      }
+      break;
+    }
+    case TransactionType.repay: {
+      mock.marketsData[Underlying.weth].poolLiquidity = constants.Zero;
+      mock.userMarketsData[Underlying.dai].scaledCollateral =
+        parseUnits("1000"); //Borrow capacity shouldn't be a problem
+      mock.userMarketsData[Underlying.weth].walletBalance = reverse
+        ? BASE_AMOUNT
+        : BASE_AMOUNT.mul(2);
+      mock.userMarketsData[Underlying.weth].scaledBorrowOnPool = reverse
+        ? BASE_AMOUNT
+        : BASE_AMOUNT.mul(2);
+
+      if (reverse && relation === OperationRelation.bigger) {
+        mock.marketsData[Underlying.weth].poolLiquidity = DELTA;
+      }
+
+      break;
+    }
+    case TransactionType.supply: {
+      mock.userMarketsData[Underlying.weth].walletBalance = reverse
+        ? BASE_AMOUNT
+        : BASE_AMOUNT.mul(2);
+
+      if (reverse && relation === OperationRelation.bigger) {
+        mock.userMarketsData[Underlying.weth].scaledSupplyOnPool = DELTA;
+      }
+      break;
+    }
+    case TransactionType.supplyCollateral: {
+      mock.userMarketsData[Underlying.dai].walletBalance = reverse
+        ? BASE_AMOUNT
+        : BASE_AMOUNT.mul(2);
+
+      if (reverse && relation === OperationRelation.bigger) {
+        mock.userMarketsData[Underlying.dai].scaledCollateral = DELTA;
+      }
+      break;
+    }
+  }
+  return mock;
+};
+
+describe("Bulker should merge", () => {
+  const userAddress = Wallet.createRandom().address;
+  let bulkerHandler: BulkerTxHandler;
+  let adapter: MorphoAaveV3Adapter;
+
+  afterEach(() => {
+    bulkerHandler.close();
+  });
+
+  [
+    TransactionType.supply,
+    TransactionType.borrow,
+    TransactionType.supplyCollateral,
+  ].forEach((baseTxSide) => {
+    [false, true].forEach((baseReverse) => {
+      [true, false].forEach((baseMax) => {
+        describe(`a ${getTxType(baseTxSide, baseReverse)}${
```

Comment:
you can leverage on a `describe.each` for a better formatting of the tests

Reply:
As far as I remember I tried but i had some limitations